workers/unity/Assets/Playground/Scripts/MonoBehaviours/PlayerColor.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Generated.Playground;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEngine;
+using Color = UnityEngine.Color;
+
+namespace Playground.MonoBehaviours
+{
+    public class PlayerColor : MonoBehaviour
+    {
+        private static MaterialPropertyBlock basicMaterial;
+        private static MaterialPropertyBlock runningMaterial;
+        private static MaterialPropertyBlock flashingMaterial;
+
+        [RuntimeInitializeOnLoadMethod]
+        public static void SetupColors()
+        {
+            basicMaterial = new MaterialPropertyBlock();
+            basicMaterial.SetColor("_Color", Color.white);
+
+            flashingMaterial = new MaterialPropertyBlock();
+            flashingMaterial.SetColor("_Color", Color.red);
+
+            runningMaterial = new MaterialPropertyBlock();
+            runningMaterial.SetColor("_Color", Color.yellow);
+        }
+
+        [SerializeField] private float flashTime = 0.05f;
+
+        [Require] PlayerInput.Requirables.Reader reader;
+
+        private MeshRenderer renderer;
+        private bool isRunning = false;
+        private bool isFlashing = false;
+        private float flashStartTime;
+
+        void OnEnable()
+        {
+            if (reader != null) // TODO remove in UTY-791
+            {
+                reader.ComponentUpdated += OnUpdateReceived;
+                reader.RunningUpdated += OnRunningChanged;
+            }
+        }
+
+        void OnDisable()
+        {
+            if (reader != null) // TODO remove in UTY-791
+            {
+                reader.ComponentUpdated -= OnUpdateReceived;
+                reader.RunningUpdated -= OnRunningChanged;
+            }
+        }
+
+        private void Awake()
+        {
+            renderer = transform.Find("Shape").GetComponent<MeshRenderer>();
```

Comment:
Needs null checks - both on the `Find` and the `GetComponent<>` calls.

Alternative, use null conditional operators and lose a bit of debug information

```
renderer = transform.Find("Shape")?.GetComponent<MeshRenderer>();
if (renderer == null)
{
	// Either the Find or GetComponent failed
}
```